### PR TITLE
Feature/dashboard

### PR DESCRIPTION
**Pull request by nasir-momin**
## Description

- This PR added 2 new screen, Dashboard where user can see the Enter time entry for the week (Not yet full completed)
- Another Project list where user can see list of project & its is searchable.
- Along with a new Tab view at bottom for user to navigate between Time sheet screen, Report screen & Account screen

## Type of Change

- [x ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

Simulator Screen Shot - iPhone 11 Pro - 2022-06-09 at 19 03 09

Simulator Screen Shot - iPhone 11 Pro - 2022-06-09 at 19 03 18

### CODE REVIEW

File: iosApp/iosApp/Reusable Common Components/Utils/DateHandler.swift [hunk 1]
```diff
@@ -0,0 +1,40 @@
+//
+//  DateHandler.swift
+//  iosApp
+//
+//  Created by Nasir Ahmed Momin on 08/06/22.
+//  Copyright © 2022 Mutual Mobile. All rights reserved.
+
+///  Inspired by Mutual Mobile's Great developer `Vaibhav Khatri` from his talk on `Property Wrapper`
```
Comment (chitvanMM) [inline iosApp/iosApp/Reusable Common Components/Utils/DateHandler.swift:8]: 🏆

File: iosApp/iosApp/Views/Dashboard/ProjectListView.swift [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            .navigationTitle(Text("Projects"))
+        }
+    }
+    
+    private var searchedResult: [String] {
+        if searchText.isEmpty {
+            return projects
+        } else {
+            return projects.filter { $0.contains(searchText) }
+        }
```
Comment (chitvanMM) [inline iosApp/iosApp/Views/Dashboard/ProjectListView.swift:55]: move this to VM

File: iosApp/iosApp/Views/Dashboard/TimeSheetView.swift [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+                                   sunday: 0.0,
+                                   monday: 3.0,
+                                   tuesday: 8.00,
+                                   wednesday: 8.0,
+                                   thursday: 4.30,
+                                   friday: 1.2)
+    }
+    
+    func validate(for selectedDay: WeekLogHour.Day) {
+        objectWillChange.send()
```
Comment (chitvanMM) [inline iosApp/iosApp/Views/Dashboard/TimeSheetView.swift:32]: Why call objectWillChange and not use a published property?

File: iosApp/iosApp/Views/Dashboard/TimeSheetView.swift [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+import SwiftUI
+
+struct TimeSheetView_Previews: PreviewProvider {
+    static var previews: some View {
+        TimeSheetView()
+    }
+}
+
+
+class TimeSheetStore: ObservableObject {
```
Comment (chitvanMM) [inline iosApp/iosApp/Views/Dashboard/TimeSheetView.swift:18]: Move TimeSheetStore to a separate file

File: iosApp/iosApp/Views/Dashboard/TimeSheetView.swift [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+        } else {
+            self.logHour.selectedDay = selectedDay
+            self.logHour.today = nil
+        }
+    }
+}
+
+struct TimeSheetView: View {
+    
+    @ObservedObject var store = TimeSheetStore()
```
Comment (chitvanMM) [inline iosApp/iosApp/Views/Dashboard/TimeSheetView.swift:46]: Prefer StateObject over ObservedObject

File: iosApp/iosApp/Views/Dashboard/TimeSheetView.swift [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+    
+    @State private var calenderTapped = false
+    @State private var infoTapped = false
+    @State private var moreTapped = false
+    @State private var presentTimeEntrySheet = false
```
Comment (chitvanMM) [inline iosApp/iosApp/Views/Dashboard/TimeSheetView.swift:51]: Can these properties be part of your store?

File: iosApp/iosApp/Views/Dashboard/TimeSheetView.swift [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+    
+    @DateHandler private var headerDate = Date()
+    
+    var body: some View {
+        VStack(alignment: .center) {
+            headerView
+            Spacer()
+            ContainerView(presentTimeEntrySheet: $presentTimeEntrySheet, quoteText: "Here goes some of great quote from data model")
```
Comment (chitvanMM) [inline iosApp/iosApp/Views/Dashboard/TimeSheetView.swift:59]: break params into separate lines.

File: iosApp/iosApp/Views/Dashboard/TimeSheetView.swift [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+                    Image(systemName: "info.circle")
+                }
+                .buttonStyle(BorderlessButtonStyle())
+                .padding(.trailing)
+                
+                Button {
+                    moreTapped = true
+                } label: {
+                    Image(systemName: "ellipsis")
+                }
```
Comment (chitvanMM) [inline iosApp/iosApp/Views/Dashboard/TimeSheetView.swift:93]: try placing all buttons inside GroupView {...} and then apply 
`.buttonStyle(BorderlessButtonStyle())`
` .padding(.trailing)`

and see if that works as expected

File: iosApp/iosApp/Views/Dashboard/TimeSheetView.swift [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+            WeekView(logHours: store.logHour) { selectedDay in
+                store.validate(for: selectedDay)
+            }
+        }
+        .padding(.vertical)
+        .background(.quaternary)
+    }
+}
+
+struct ContainerView: View {
```
Comment (chitvanMM) [inline iosApp/iosApp/Views/Dashboard/TimeSheetView.swift:107]: Please use a better name for this View, not able to guess what this view is about from current name

File: iosApp/iosApp/Views/Dashboard/TimeSheetView.swift [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+    
+    @Binding var presentTimeEntrySheet: Bool
+    var quoteText: String
```
Comment (chitvanMM) [inline iosApp/iosApp/Views/Dashboard/TimeSheetView.swift:110]: change to let, if possible

File: iosApp/iosApp/Views/LoginView.swift [hunk 11]
```diff
@@ -18,7 +18,8 @@ class AuthStore: ObservableObject {
 }
 
 struct LoginView: View {
-    
+    @EnvironmentObject var rootStore: RootStore
+
     @ObservedObject private var store = AuthStore()
```
Comment (chitvanMM) [inline iosApp/iosApp/Views/LoginView.swift:23]: prefer StateObject over ObservedObject

File: iosApp/iosApp/Views/LoginView.swift [hunk 12]
```diff
@@ -141,19 +142,19 @@ struct LoginView: View {
...
         
         loginDataModel.login(email: email, password: password)
         
-        loginDataModel.praxisCommand = { command in
-            print("command \(command)  \(type(of: command)) ")
-            if let navigationCommand = (command as? NavigationPraxisCommand) {
-                print("command .route \(navigationCommand.screen) \(navigationCommand.component1())  \(navigationCommand.route) ")
-            }
-        }
+//        loginDataModel.praxisCommand = { command in
```
Comment (chitvanMM) [inline iosApp/iosApp/Views/LoginView.swift:152]: remove commented code or add a TODO

File: iosApp/iosApp/Views/RootView.swift [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+//  RootView.swift
+//  iosApp
+//
+//  Created by Nasir Ahmed Momin on 07/06/22.
+//  Copyright © 2022 Mutual Mobile. All rights reserved.
+//
+
+import SwiftUI
+
+class RootStore: ObservableObject {
```
Comment (chitvanMM) [inline iosApp/iosApp/Views/RootView.swift:11]: Move store to a separate file

File: iosApp/iosApp/iOSApp.swift [hunk 14]
```diff
@@ -6,7 +6,7 @@ struct iOSApp: App {
 
 	var body: some Scene {
 		WindowGroup {
-            LaunchView()
+            RootView()
```
Comment (chitvanMM) [inline iosApp/iosApp/iOSApp.swift:9]: ctrl + i, alignment issue

**Review by chitvanMM — COMMENTED** [review]
Changes look pretty good 👍  
I have suggested some minor changes, feel free to reach out to me in case of any discussion is required.

File: iosApp/iosApp/Reusable Common Components/Utils/DateHandler.swift [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+
+import Foundation
+
+@propertyWrapper
+struct DateHandler {
+    private var _date: Date
```
[the remark this answers is not in the document]
Comment (nasir-momin) [reply iosApp/iosApp/Reusable Common Components/Utils/DateHandler.swift:14]: Thanks, Made changes, will be visible in next PR

File: iosApp/iosApp/Reusable Common Components/Utils/DateHandler.swift [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+    @StringTrimmer var format: String
+        
+    init(wrappedValue: Date, format: String = "EEEE, dd MMM") {
```
[the remark this answers is not in the document]
Comment (nasir-momin) [reply iosApp/iosApp/Reusable Common Components/Utils/DateHandler.swift:17]: Thanks, Made changes, will be visible in next PR

File: iosApp/iosApp/Reusable Common Components/Utils/StringTrimmer.swift [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+//
+//  Created by Nasir Ahmed Momin on 08/06/22.
+//  Copyright © 2022 Mutual Mobile. All rights reserved.
+
+///  Inspired by Mutual Mobile's Great developer `Vaibhav Khatri` from his talk on `Property Wrapper`
+
+import Foundation
+
+@propertyWrapper
+struct StringTrimmer {
```
[the remark this answers is not in the document]
Comment (nasir-momin) [reply iosApp/iosApp/Reusable Common Components/Utils/StringTrimmer.swift:13]: Thanks, Made changes, will be visible in next PR

File: iosApp/iosApp/Reusable Common Components/Views/WeekView.swift [hunk 18]
```diff
@@ -0,0 +1,205 @@
...
+    
+    var body: some View {
+        HStack(alignment: .center) {
+            ForEach(logHours.days) { day in
+                Button {
+                    daySelection(day)
+                } label: {
+                    VStack {
+                        Text(day.tuple.name)
+                            .padding(EdgeInsets(top: verticalPadding,
```
[the remark this answers is not in the document]
Comment (nasir-momin) [reply iosApp/iosApp/Reusable Common Components/Views/WeekView.swift:54]: Thanks, Nice idea, Updated will be available in next PR

File: iosApp/iosApp/Views/LoginView.swift (hunk 12 above)
Comment (nasir-momin) [reply iosApp/iosApp/Views/LoginView.swift:152]: Removed in next branch, will be available in next MR

File: iosApp/iosApp/Views/Dashboard/TimeSheetView.swift (hunk 10 above)
Comment (nasir-momin) [reply iosApp/iosApp/Views/Dashboard/TimeSheetView.swift:110]: Thanks, Made changes, will be visible in next PR

File: iosApp/iosApp/iOSApp.swift (hunk 14 above)
Comment (nasir-momin) [reply iosApp/iosApp/iOSApp.swift:9]: Available in next PR

**Pull request merged**